apps/04TowersOfHanoi.js

```diff
@@ -13,34 +13,62 @@ var stacks = {
   c: []
 };
 
+// resets stacks
+function startOver() {
+  stacks = {
+    a: [4, 3, 2, 1],
+    b: [],
+    c: []
+  }
+};
+
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  // Pops from start stack and then pushes onto end stack
+  stacks[endStack].push(stacks[startStack].pop());
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
   // Your code here
+  var endPieceFrom = stacks[startStack][stacks[startStack].length - 1];
+  var endPieceTo = stacks[endStack][stacks[endStack].length - 1];
+  if (endPieceFrom < endPieceTo || endPieceTo === undefined) {
+    movePiece(startStack, endStack);
```

Comment:
We probably don't want to be moving pieces as part of the isLegal function; this would fit better in the towersOfHanoi function